src/main/java/com/google/javascript/gents/CommentLinkingPass.java

```diff
@@ -223,12 +291,14 @@ public final boolean shouldTraverse(NodeTraversal nodeTraversal, Node n, Node pa
       } else if (getLastLineOfCurrentComment() == line - 1) {
         // Comment ends just before code
         linkCommentBufferToNode(n);
-      } else if (!hasRemainingComments()) {
-        // Exhausted all comments, output floating comment before current node
```

Comment:
Discussed offline. This is OK for a bug fix.
In general this is confusing here. A lot of invariants are hidden in code. We should consider refactoring if we encounter more issues with comments.